src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -635,14 +749,36 @@ public Object correct(Object value, Object def)
         @Override
         public String toString()
         {
-            if (clazz == Integer.class) {
-                if (max.equals(Integer.MAX_VALUE)) {
-                    return "> " + min;
-                } else if (min.equals(Integer.MIN_VALUE)) {
-                    return "< " + max;
-                }
-            } // TODO add more special cases?
-            return min + " ~ " + max;
+            final String greaterThan = "> " + min;
+            final String lessThan = "< " + max;
+            final String between = min + " ~ " + max;
+
+            if (clazz == Byte.class)
+                if (max.equals(Byte.MAX_VALUE)) return greaterThan;
+                else if (min.equals(Byte.MIN_VALUE)) return lessThan;
+                else return between;
+            if (clazz == Short.class)
+                if (max.equals(Short.MAX_VALUE)) return greaterThan;
+                else if (min.equals(Short.MIN_VALUE)) return lessThan;
+                else return between;
+            if (clazz == Integer.class)
+                if (max.equals(Integer.MAX_VALUE)) return greaterThan;
+                else if (min.equals(Integer.MIN_VALUE)) return lessThan;
+                else return between;
+            if (clazz == Long.class)
+                if (max.equals(Long.MAX_VALUE)) return greaterThan;
+                else if (min.equals(Long.MIN_VALUE)) return lessThan;
+                else return between;
+            else if (clazz == Float.class)
+                if (max.equals(Float.MAX_VALUE)) return greaterThan;
+                else if (min.equals(Float.MIN_VALUE)) return lessThan;
```

Comment:
I believe this should be `-Float.MAX_VALUE` one thing I discovered relatively recently is that `Float.MIN_VALUE` represents the smallest positive value a float can represent, and not the smallest value that fits in a float (which is represented by `-Float.MAX_VALUE`).

Reply:
I'm assuming most people would be using `defineInRange("aFloat/aDouble", defaultValue, Float/Double.MIN_VALUE, Float/Double.MAX_VALUE);` though?

Reply:
For Mekanism Tools, where I am defining attack speed I have negative values (given how vanilla treats attack speed). It also doesn't help that iirc `0 < Float.MIN_VALUE`.

Reply:
So I shouldn't be using `Float/Double.MIN_VALUE`.
Should I use `POSITIVE_INFINITY`/`NEGATIVE_INFINITY` or `-Float/Double.MAX_VALUE`?

Reply:
Also I am not sure and I do believe it is unlikely they will be using `-Float/Double.MAX_VALUE` as the final bounds of their range, but I do think given how you are handling integers, longs, and shorts that it is consistent between them all.

Reply:
I can compare against both `MAX_VALUE` & `POSITIVE_INFINITY` and `-MAX_VALUE` and `NEGATIVE_INFINITY`. Should I do this or just `MAX_VALUE` and `-MAX_VALUE`?

Reply:
Personally I think just `MAX_VALUE` and `-MAX_VALUE` is enough, but I will leave this for other people to give their input on.

Reply:
tterag on discord did also mention he agrees that it should be `-MAX_VALUE`, when I asked other people to take a look.

Reply:
Will change it to `-MAX_VALUE`